homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +81,112 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
-    sensors = [
-        DwdWeatherWarningsSensor(api, name, condition)
-        for condition in config[CONF_MONITORED_CONDITIONS]
-    ]
+    # Build sensor list and activate update only for the first one
+    sensors = []
+    call_update = True
+    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        sensors.append(DwdWeatherWarningsSensor(api, name, sensor_type, call_update))
+        call_update = False
 
     add_entities(sensors, True)
 
 
 class DwdWeatherWarningsSensor(Entity):
     """Representation of a DWD-Weather-Warnings sensor."""
 
-    def __init__(self, api, name, variable):
+    def __init__(self, api, name, sensor_type, call_update):
         """Initialize a DWD-Weather-Warnings sensor."""
         self._api = api
         self._name = name
-        self._var_id = variable
-
-        variable_info = MONITORED_CONDITIONS[variable]
-        self._var_name = variable_info[0]
-        self._var_units = variable_info[1]
-        self._var_icon = variable_info[2]
+        self._sensor_type = sensor_type
+        self._call_update = call_update
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._name} {self._var_name}"
+        return f"{self._name} {MONITORED_CONDITIONS[self._sensor_type][0]}"
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        return self._var_icon
+        return MONITORED_CONDITIONS[self._sensor_type][2]
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        return self._var_units
+        return MONITORED_CONDITIONS[self._sensor_type][1]
 
     @property
     def state(self):
         """Return the state of the device."""
-        try:
-            return round(self._api.data[self._var_id], 2)
-        except TypeError:
-            return self._api.data[self._var_id]
+        if self._sensor_type == CURRENT_WARNING_SENSOR:
+            return self._api.current_warning_level
+        return self._api.expected_warning_level
 
     @property
     def device_state_attributes(self):
         """Return the state attributes of the DWD-Weather-Warnings."""
-        data = {ATTR_ATTRIBUTION: ATTRIBUTION, "region_name": self._api.region_name}
-
-        if self._api.region_id is not None:
-            data["region_id"] = self._api.region_id
-
-        if self._api.region_state is not None:
-            data["region_state"] = self._api.region_state
-
-        if self._api.data["time"] is not None:
-            data["last_update"] = dt_util.as_local(
-                dt_util.utc_from_timestamp(self._api.data["time"] / 1000)
-            )
+        data = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_REGION_NAME: self._api.warncell_name,
+            ATTR_REGION_ID: self._api.warncell_id,
+        }
+
+        if self._api.last_update is not None:
+            data[ATTR_LAST_UPDATE] = dt_util.as_local(self._api.last_update)
```

Comment:
All timestamps in the state machine must be UTC time, not local time.

Reply:
I changed all timestamps to UTC.
For some people this will be a breaking change because the old sensor returned timestamps as local time. I felt over it myself outputting the e.g. "warning start" with `strftime` in Jinja without using `timestamp_local` and `timestamp_custom`.

Reply:
Please mention this in the breaking change paragraph in the PR description.